qiskit/quantum_info/states/statevector.py

```diff
@@ -113,6 +113,30 @@ def __init__(
                 raise QiskitError("Invalid input: not a vector or column-vector.")
         super().__init__(op_shape=OpShape.auto(shape=shape, dims_l=dims, num_qubits_r=0))
 
+    @classmethod
+    def from_circuit(cls, circuit, input_state=None):
+        """Create a Statevector from a quantum circuit.
+
+        Args:
+            circuit (QuantumCircuit): A quantum circuit
+            input_state (Statevector, optional): Input statevector. If None,
+                defaults to the zero state.
```

Comment:
I think we should probably remove this argument to keep this interface smaller. It's pretty straightforward if the end user wants to do this that they can simply call `input.evolve(qc)` or `input.evolve(Statevector.from_circuit(qc))` themselves. Adding extra arguments here will just increase the complexity and maintenance surface longer term so it would be better to just leave this signature simpler.